tinyformat.h

```diff
@@ -955,23 +956,40 @@ inline void vformat(std::ostream& out, const char* fmt, FormatListRef list)
 #ifdef TINYFORMAT_USE_VARIADIC_TEMPLATES
 
 /// Format list of arguments to the stream according to given format string.
+/// This honors the stream's existing locale conventions.
 template<typename... Args>
 void format(std::ostream& out, const char* fmt, const Args&... args)
 {
     vformat(out, fmt, makeFormatList(args...));
 }
 
 /// Format list of arguments according to the given format string and return
-/// the result as a string.
+/// the result as a string, using classic "C" locale conventions (e.g.,
+/// using '.' as decimal mark).
 template<typename... Args>
 std::string format(const char* fmt, const Args&... args)
 {
     std::ostringstream oss;
+    oss.imbue (std::locale::classic());  // force "C" locale with '.' decimal
```

Comment:
I'm a little uneasy about interfering here with the default behaviour of using the global locale.  But it's certainly more convenient than having to pass the desired locale as a parameter.

Reply:
The alternative design is for the unadorned format() to use the native locale (as before) and have users who want the C locale use the new variety and explicitly pass `std::locale::classic()`. But that seems error-prone to me, since most users would probably prefer to always have '.' and (like me) be shocked that this wasn't already the case.

Reply:
This is a tough one as there's two quite valid but competing use cases.

* For machine readable output, consistency is super important.  Using the `classic()` locale is the right thing for this.  It's also generally more consistent between environments which is great.
* When generating content for users it's arguably be more correct - and consistent with the other `format` functions - to use the current global locale for `format()`.

One way out of the conundrum might just be to add a new `cformat` function ("format in C locale"), and leave `format` as it is for consistency with the other flavours of `format`.  Obviously a better name than `cformat` would be good...